system/views/general/_openGraphMeta.cfm

```diff
@@ -14,7 +15,7 @@
 <cfoutput>
 	<meta property="og:title" content="#XmlFormat( local.title )#" />
 	<meta property="og:type"  content="#local.ogType#" />
-	<meta property="og:url"   content="#event.getSiteUrl()##HtmlEditFormat( event.getCurrentUrl() )#" />
+	<meta property="og:url"   content="#reReplace( local.ogUrl, "/([a-z][a-z]/)(\1)+", "/\1")#" />
```

Comment:
I think this approach is problematic:

1. It is only _patching_ the problem, and the business logic of that patch is taking place in a _view_
2. What is to say a user does not setup a structure themselves like /ca/fb/my-page.html - however unlikely that may be, when it does happen, the produced bug will have people scratching their heads

I think this should be reworked to get to the core of the problem. i.e. why is the language code in there twice. Fix that problem at the source, rather than patching it afterwards.

Does that make sense?

Reply:
Hi @DominicWatson,

Absolutely right. I should have mentioned that `getSiteUrl` and `getCurrentUrl` both return language slugs thus the double. I have amended it to use `event.getSiteUrl( includeLanguageSlug=false )` instead which solves the issue. There was also the option of using `getBaseUrl()` but I wasn't sure if that would break anything so I opted for the 'safer' route.

Thanks.

Reply:
Question: Is it in general correct, that `getSiteUrl` is returning the language? I think it should only return the protocol+domain+url path but not the language. Preside is redirecting to the language automatically.

Reply:
I think it is correct @madmikede and changing it would be compatibility breaking. As Bryan points out, you are able to override this behaviour should you wish.